lib/msal-angular/src/msal.interceptor.ts

```diff
@@ -11,56 +11,48 @@ import 'rxjs/add/operator/mergeMap'
-                    Authorization: `Bearer ${tokenStored.token}`,
-                }
-            });
-            return next.handle(req).do(event => {}, err => {
-                if (err instanceof HttpErrorResponse && err.status == 401) {
-                    var scopes = this.auth.getScopesForEndpoint(req.url);
-                    var tokenStored = this.auth.getCachedTokenInternal(scopes);
-                    if (tokenStored && tokenStored.token) {
-                        this.auth.clearCacheForScope(tokenStored.token);
-                    }
-                    var msalError = new MSALError(JSON.stringify(err), "", JSON.stringify(scopes));
-                    this.broadcastService.broadcast('msal:notAuthorized', msalError);
-                }
-            });
-        }
-        else {
-            return Observable.fromPromise(this.auth.acquireTokenSilent(scopes).then(token => {
-                const JWT = `Bearer ${token}`;
-                return req.clone({
-                    setHeaders: {
-                        Authorization: JWT,
-                    },
-                });
-            })).mergeMap(req => next.handle(req).do(event => {}, err => {
-                if (err instanceof HttpErrorResponse && err.status == 401) {
-                    var scopes = this.auth.getScopesForEndpoint(req.url);
-                    var tokenStored = this.auth.getCachedTokenInternal(scopes);
-                    if (tokenStored && tokenStored.token) {
-                        this.auth.clearCacheForScope(tokenStored.token);
+
+        let accessToken: string;
+
+        // Acquire a token for this request, and attach as proper auth header.
+        return Observable.fromPromise(
+            this.auth.acquireTokenSilent({ scopes })
+                .then((response: AuthResponse) => {
+                    accessToken = response.accessToken;
+                    const authHeader = `Bearer ${response.accessToken}`;
+                    return req.clone({
+                        setHeaders: {
+                            Authorization: authHeader,
+                        }
+                    });
+                })
+        ).mergeMap(nextReq => {
+            // Call next handler, and if that errors, broadcast error
+            return next.handle(nextReq).do(
+                event => {},
+                err => {
+                    if (err instanceof HttpErrorResponse && err.status == 401) {
+                        this.auth.clearCacheForScope(accessToken);
+                        this.broadcastService.broadcast('msal:notAuthorized', err.message);
                     }
-                    var msalError = new MSALError(JSON.stringify(err), "", JSON.stringify(scopes));
-                    this.broadcastService.broadcast('msal:notAuthorized', msalError);
-                }
-            })); //calling next.handle means we are passing control to next interceptor in chain
-        }
+
+                    throw err;
```

Comment:
TODO: remove this?

Reply:
If you remove this won't errors just get swallowed by this handler?

Reply:
It will get broadcast as `msal:notAuthorized`.

Reply:
I'll gather feedback on this, but for will keep as-is.